File: lib/commands/builtin/focus.ts
import type { CommandSpec } from "../types"

export const focusCommand: CommandSpec = {
  name: "focus",
  summary: "show last focused window tracking",
  usage: ["focus"],
  man: [
    "NAME",
    "  focus - show last focused window tracking",
    "",
    "SYNOPSIS",
    "  focus",
    "",
    "NOTE",
    "  Used when tab id is omitted for back / forward."
  ],
  async execute(ctx) {
    return ctx.focusInfo()
  }
}

File: wasm/bmxt-core/src/cmd/focus.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "focus",
    aliases: &[],
    usage_primary: "focus",
    man: &[
        "NAME",
        "  focus - show last focused window tracking",
        "",
        "SYNOPSIS",
        "  focus",
        "",
        "NOTE",
        "  Used when tab id is omitted for back / forward.",
    ],
};

pub fn run(_args: &[String]) -> DispatchJson {
    DispatchJson::effects(vec![Effect::FocusInfo])
}
